webclient/src/forms/CardImportForm/CardImportForm.tsx

```diff
@@ -0,0 +1,169 @@
+        setImportedCards(cards);
+        setImportedSets(sets);
+
+        handleNext();
+      })
+      .catch(e => {
+        // TODO: render error in UI
+        console.log(e);
+      });
+  }
+
+  const handleCardSave = async () => {
+    await CardDTO.bulkAdd(importedCards);
+    await SetDTO.bulkAdd(importedSets);
+
+    handleNext();
+  }
+
+  const handleTokenDownload = ({ tokenDownloadUrl }) => {
+    cardImporterService.importTokens(tokenDownloadUrl)
+      .then(async tokens => {
+        await TokenDTO.bulkAdd(tokens);
+        handleNext();
+      })
+      .catch(e => {
+        // TODO: render error in UI
+        console.log(e);
+      });
+  }
+
+  const BackButton = () => (
+    <Button onClick={handleBack}>Go Back</Button>
+  );
+
+  const getStepContent = (stepIndex) => {
+    switch (stepIndex) {
+      case 0: return (
+        <Form className="cardImportForm" onSubmit={handleSubmit(handleCardDownload)}>
+          <div className="cardImportForm-item">
+            <Field label="Download URL" name="cardDownloadUrl" component={InputField} />
+          </div>
+
+          <Button color="primary" type="submit">Import</Button>
+        </Form>
+      );
+
+      case 1: return (
+        <div>
+          <div>
+            <CardsImported cards={importedCards} sets={importedSets} />
+          </div>
+          <BackButton />
+          <Button color="primary" onClick={handleCardSave}>Save</Button>
+        </div>
+      );
+
+      case 2: return (
+        <Form className="cardImportForm" onSubmit={handleSubmit(handleTokenDownload)}>
+          <div className="cardImportForm-item">
+            <Field label="Download URL" name="tokenDownloadUrl" component={InputField} />
+          </div>
+
+          <BackButton />
+          <Button color="primary" type="submit">Import</Button>
+        </Form>
+      );
+
+      case 3: return (
+        <div>
+          <div>Finished!</div>
+
+          <BackButton />
+          <Button color="primary" onClick={onClose}>Done</Button>
+        </div>
+      );
+    }
+  };
+
+  return (
+    <div>
+      <Stepper activeStep={activeStep} alternativeLabel>
+        {steps.map((label) => (
+          <Step key={label}>
+            <StepLabel>{label}</StepLabel>
+          </Step>
+        ))}
+      </Stepper>
+
+      <div>{ getStepContent(activeStep) }</div>
+    </div>
+  );
+}
+
+const CardsImported = ({ cards, sets }) => {
+  const items = [
+    ...sets.map(set => (
+      <div>{set.name}: {set.cards.length} cards imported</div>
+    ) ),
+
+    (
+      <div>
+        <strong>Import finished: {cards.length} cards.</strong>
+      </div>
+    )
+  ];
+
+  return (
+    <div className="card-import-list">
+      <ScrollToBottomOnChanges changes={sets} content={(
+        <VirtualList
+          itemKey={(index) => index }
+          items={items}
+          size={15}
+        />
+      )} />
+    </div>
+  );
+};
+
+const propsMap = {
+  form: FormKey.CARD_IMPORT,
+  onClose: Function
```

Comment:
I couldn't figure out how to get this onClose method passed in from the parent dialog component. You'll notice above I've hijacked the onSubmit method that i'm able to pass in.

```
const { handleSubmit, onSubmit:onClose } = props;
```